layouts/partials/scripts.html

```diff
@@ -102,4 +102,26 @@
 {{ end -}}
 
 <script src='{{ "js/tabpane-persist.js" | relURL }}'></script>
+
+{{- if and .Site.Params.search (isset .Site.Params.search "algolia") -}}
+  {{ template "algolia/scripts" .Site.Params.search.algolia -}}
+{{ end -}}
+
 {{ partial "hooks/body-end.html" . -}}
+
+{{ define "algolia/scripts" }}
```

Comment:
Same as for css: why not retrieving the .js resource at build time?

```suggestion
{{ define "algolia/scripts" }}
  {{ $version := "v3.5.1" -}}
  {{ with .Site.Params.search.algolia.version -}}
    {{ $version = . -}}
  {{ end -}}
  {{ $cdnurl := printf "https://cdn.jsdelivr.net/npm/@docsearch/js@%s" $version -}}
  {{ if eq (resources.GetRemote $cdnurl) nil -}}
    {{ errorf "Invalid Algolia version %s, could not retrieve this version from CDN" $version -}}
  {{ end -}}

  {{ with resources.GetRemote $cdnurl -}}
    {{ with .Err -}}
      {{ errorf "Error while retrieving resource %q from CDN. Status: %s." $cdnurl .Data.Status -}}
    {{ else -}}
      {{ $secureJS := . | resources.Fingerprint "sha512" -}}
<script src="{{ .RelPermalink }}" integrity="{{ $secureJS.Data.Integrity }}"
      crossorigin="anonymous" ></script>
    {{ end -}}
  {{ end -}}
```